### PR TITLE
898 oldaccinputhash for lower forkids

### LINKED ISSUE #898: OldAccInputHash for lower forkids
The contract ABI is different for lower forkids, so calling to get oldAccInputHash from the L1 for lower forkids fails. This needs a little investigation and an implementation to fix.

Currently getBatchByNumber is returning the oldAccInputHash as 0x00...0 when retries are exceeded and this fails, which needs a proper fix.

### PR DESCRIPTION
**Pull request by V-Staykov**

### CODE REVIEW

File: cmd/rpcdaemon/commands/zkevm_api.go [hunk 1]
```diff
@@ -695,12 +694,11 @@ func (api *ZkEvmAPIImpl) GetBatchByNumber(ctx context.Context, batchNumber rpc.B
...
 
-	oldAccInputHash, err := api.l1Syncer.GetOldAccInputHash(ctx, &api.config.AddressRollup, api.config.L1RollupId, batchNo)
+	accInputHash, err := api.getAccInputHash(ctx, hermezDb, batchNo)
 	if err != nil {
-		log.Warn("Failed to get old acc input hash", "err", err)
-		batch.AccInputHash = common.Hash{}
+		return nil, fmt.Errorf("failed to get acc input hash for batch %d: %w", batchNo, err)
 	}
-	batch.AccInputHash = oldAccInputHash
+	batch.AccInputHash = *accInputHash
```
Comment (hexoscott) [inline cmd/rpcdaemon/commands/zkevm_api.go]: worth a nil check here to avoid a potential panic?

File: cmd/rpcdaemon/commands/zkevm_api.go [hunk 2]
```diff
@@ -1067,7 +1149,7 @@ func (api *ZkEvmAPIImpl) GetProverInput(ctx context.Context, batchNumber uint64,
 		return nil, err
 	}
 
-	oldAccInputHash, err := api.l1Syncer.GetOldAccInputHash(ctx, &api.config.AddressRollup, api.config.L1RollupId, batchNumber)
+	oldAccInputHash, err := api.getAccInputHash(ctx, hDb, batchNumber)
```
Comment (hexoscott) [inline cmd/rpcdaemon/commands/zkevm_api.go]: as this is "old" acc input hash, should it be for the previous batch/sequence?

File: cmd/rpcdaemon/commands/zkevm_api.go (hunk 1 above)
Comment (V-Staykov) [reply cmd/rpcdaemon/commands/zkevm_api.go]: sure!

File: cmd/rpcdaemon/commands/zkevm_api.go (hunk 2 above)
Comment (V-Staykov) [reply cmd/rpcdaemon/commands/zkevm_api.go]: not sure about this one, maybe @kstoykov knows if we need the current batch accInputHash here as well, or it is the previous sequence's one?

Comment (V-Staykov) [reply cmd/rpcdaemon/commands/zkevm_api.go]: Actually so far this was just zeroed hash, so it won't brake anything. :D We should maybe later check if this should be for the current batch or not.

Comment (hexoscott) [reply cmd/rpcdaemon/commands/zkevm_api.go]: We could compare a local node with a legacy node for one of the integration networks and check if they agree?  Would be good to nail it first time if we can 💪

Comment (V-Staykov) [reply cmd/rpcdaemon/commands/zkevm_api.go]: Spoke with Carlos and he said this should be for previous batch. I made the change

**Review by kstoykov — DISMISSED** [review]

**Review by kstoykov — APPROVED** [review]

**Pull request merged**